src/mpid/ch4/netmod/ucx/ucx_recv.h

```diff
@@ -37,7 +37,14 @@ static inline void MPIDI_UCX_recv_cmpl_cb(void *request, ucs_status_t status,
 
     if (unlikely(status == UCS_ERR_CANCELED)) {
         MPIR_STATUS_SET_CANCEL_BIT(rreq->status, TRUE);
-    } else {
+    }
+    else if (unlikely(status == UCS_ERR_MESSAGE_TRUNCATED)) {
```

Comment:
If UCX is unable to give us reliable data, then we probably shouldn't pass it on.